client/apps/edit/components/layout/index.coffee

```diff
@@ -133,3 +136,50 @@ module.exports = class EditLayout extends Backbone.View
   toggleDragover: (e) ->
     $(e.currentTarget).closest('.dashed-file-upload-container')
       .toggleClass 'is-dragover'
+
+  getLinkableText: ->
+    fullText = @getBodyText()
+    fullText.match(/==(\S+[^==]*\S)==/ig)
+
+  autolinkText: (evt) ->
+    $('#autolink-status').addClass('searching').html('Linking...')
+    $('#edit-content__overlay').addClass('disabled')
+    linkableText = @getLinkableText()
+    # searchTypes = ['artist', 'profile', 'show', 'gene', 'city']
+    # searchQueryParam = searchTypes.map((t) -> "type[]=#{t}").join("&")
+    async.mapLimit linkableText, 5, ((findText, cb) =>
+      text = findText.split('==').join('')
+      request
+        .get("/api/search?term=#{text}")
+        .end (err, res) =>
+          if err or res.body.total < 1
+            @article.replaceLink(findText, text)
+            return cb()
+
+          valid_results = @findValidResults(res.body.hits)
+          if valid_results.length == 0
+            @article.replaceLink(findText, text)
+            return cb()
+
+          @article.replaceLink(findText, @getNewLinkFromHits(valid_results) || text)
+          return cb()
+    ), (err, result) =>
+      @article.sections.trigger 'change:autolink'
+      $('#autolink-status').removeClass('searching').html('Auto-Link')
+      $('#edit-content__overlay').removeClass('disabled')
+
+  findValidResults: (hits) ->
+    _.reject(hits, (h) -> h._score < 6.5 || h._source.visible_to_public == false)
```

Comment:
:+1: 6.5 is the threshold we use for detecting artist duplicates. You might get away with lowering it slightly for fuzzier matches, but this should work pretty well.

Reply:
I think less fuzzy works better in our case, we replace existing term with what we received from search results and i can imagine not creating the link would be better than replacing text with wrong result. I should still look into scoring system, how did we came up with 6.5?

Reply:
I think 6.5 was just by manual observation running the duplicate artist report and seeing if they were real dupe candidates that it was picking up on.